### ISSUE TITLE
Request - Artwork For Cmus?

### ISSUE THREAD

**Issue by Geyup** [cmus/cmus#495]
Hello,

Would it be possible to have cmus display artwork, or can it do this already?

As an example Ranger a console based file manager using w3m can display images.

http://ranger.nongnu.org/

**Comment by mahkoh** [comment]
Do you mean like this: http://ranger.nongnu.org/screenshots/ranger-screenshot_large-image-preview.png

Or via some external program?

**Comment by Geyup** [comment]
Yes, album art displayed in the app and like how deadbeef is with images of the album art displayed on the left side of the screen in this image below;

https://cloud.githubusercontent.com/assets/13072581/18575401/f7d3c756-7bc6-11e6-9d11-5ee53fd13601.jpg

**Comment by ferviron** [comment]
Well, since I am also very interested in this feature I am going to add some more information to it.

@mahkoh the screenshot you used as an example is indeed a way of showing the artwork within ranger in the terminal or in a framebuffer. If I am correct, ranger achieves this result with the help of `libcaca` (caca-utils) using `img2txt`, which creates a beautiful ASCII representation of the artwork.

However, what I believe @Geyup was referring to, was ranger's ability to display [_true colour_](http://ranger.nongnu.org/screenshots/ranger-screenshot-shamounni.png) images in the terminal or in the framebuffer, and this is what I am asking to include in `cmus`. This is achieved using w3m, a terminal web browser, which has the aforementioned capability. [Here](https://github.com/voidlinux/void-packages/blob/72035e6b43fa26cc0228a106a49a9c7e5db85a07/srcpkgs/w3m/template) is Void's build template for w3m, which breaks it into the browser and the image display, so it will be easier to understand how this feature works. W3m should not be needed to use `cmus` as it is done in `ranger`, which through a config option enables this feature or disables it, however, it would only work if w3m is installed with image display support (most systems build it with it, in Void's case they just decided to separate the features in different packages).

As a proposal, the artwork could be displayed under the Artist/Album section. Thank you!

**Comment by mahkoh** [comment]
I do not believe that this will ever be implemented.

**Comment by Geyup** [comment]
Sorry for late reply...

thanks

**Comment by yagarea** [comment]
w3m has great API for in terminal image preview. Filemanager ranger uses that:

![](https://k38.kn3.net/taringa/1/4/6/0/4/4/8/lordii/EEE.jpg)

**Comment by gunqqer** [comment]
There's also a rust program, `viu`, which also can do this. I have a working script to show artwork (although a biiit pixelated, gives the real command line feel to it)
```#!/bin/bash
while true; do
viu "$(urlencode -d "$(playerctl --player=cmus metadata mpris:artUrl)" | sed 's/^file:\/\///')"
done
```

**Comment by nefthy** [comment]
Terminals do not have sufficient capabilities to display images. w3m and ranger do some funky x11 hackery to draw the image over the terminal window. I do not see that going into cmus.

**Comment by gunqqer** [comment]
Yeah, tbh, while it would be nice to have it in cmus, trying to get w3m/w3mimgdisplay working is not fun at all. Not only are those tools difficult to work with, they don't even work well enough. If you try switching virtual desktops it'll get rid of the image.

Something like FBI or FIM could work better, but that would just be making a new window to display the image which defeats the purpose of having an inbuilt image. So I'd say if you want image viewing you can use my script or make your own way to view images however you want them.

**Comment by lochlainn** [comment]
@gunqqer if that script is living in a repository now, it would be really useful to link it in the [wiki](https://github.com/cmus/cmus/wiki)! I'm sure some other people would like to use it as well.

**Comment by gunqqer** [comment]
@lochlainn Well, here ya are: https://github.com/gunqqer/musicScripts
https://github.com/gunqqer/musicScripts/blob/master/viu.sh
Feel free to submit PRs if you want to improve it
